### ISSUE TITLE
[Docker] environment variables can not be matched using metavariables in the ENV statement

### ISSUE THREAD

**Issue by 0xDC0DE** [semgrep/semgrep#4866]
**Describe the bug**
I want to find the value of a environment variable declared with ENV
```
pattern: ENV ...
```
matches the line on my local version, but not in the playground. 
https://semgrep.dev/s/pieterdc1:dockerfile-env-ellipse

<img width="1649" alt="image" src="https://user-images.githubusercontent.com/3809983/159712676-d7f20809-537b-4220-a392-31c17228ee06.png">

```
pattern: ENV $KEY=$VALUE
```
doesn’t match in either place

https://semgrep.dev/s/pieterdc1:dockerfile-env-value
<img width="1651" alt="image" src="https://user-images.githubusercontent.com/3809983/159712901-98f55835-89e5-4394-9e61-1921482a9948.png">

I also found that this is not parseable when using `semgrep-core -dump_ast`
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/3809983/159712507-dc2c0687-7b17-4cb5-b06e-7162ffde082f.png">

**Expected behavior**
The most intuitive would be for environment values to be evaluated, When writing a rule that checks for the directory I was hoping this would just work:
https://semgrep.dev/s/pieterdc1:dockerfile-workdir

Otherwise, I would expect to be able to capture the environment variable name and value into two separate metavariables so I can check 
1) where the variable is used 
2) what the value of the variable is 
Something similar to this rule:
https://semgrep.dev/s/pieterdc1:dockerfile-workdir-env

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [X] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Use case**
Porting r2c rules from generic to dockerfile parser

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1032/[docker]-environment-variables-can-not-be-matched-using-metavariables.

**Comment by mjambon** [comment]
> I also found that this is not parseable when using `semgrep-core -dump_ast`

To parse a pattern, use `-dump_pattern` instead of `dump_ast`.

**Comment by mjambon** [comment]
@pieterdc1 I'll let you report this to the app team since it appears to be an issue with the live editor, possibly because the target file is named wrong (given the presence of the `include` filters).

**Comment by 0xDC0DE** [comment]
@mjambon 

You are right. The first (ellipse not parsing in the playground) is an issue with the live editor.

However, the second (metavariables not working to verify the value of an environment variable) I think is a problem with semgrep.

The example with metavariables (https://semgrep.dev/s/pieterdc1:dockerfile-env-value) does not work on my machine locally either.

<img width="933" alt="image" src="https://user-images.githubusercontent.com/3809983/159865224-398bdef3-01e1-4900-bbe0-04aaef66c966.png">

I've tried the `-dump_pattern` as you suggested. The case for the metavariables looks very different to me, and possible incorrect.?

<img width="915" alt="image" src="https://user-images.githubusercontent.com/3809983/159864544-5d0e3f0a-69bc-4157-85fd-71f631564ac0.png">

**Comment by mjambon** [comment]
I didn't realize you were reporting two different problems. Next time, please file two separate issues.

**Comment by mjambon** [comment]
ok, so the pattern `ENV $K=whatever` isn't supported at the moment as you found out. If you go to this folder https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests/dockerfile and do `git grep ENV`, you'll see the different test cases involving `ENV`. The [todo](https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests/dockerfile/todo) folder contains the known bugs. This one was tricky to implement so I left it for later.

**Comment by nbrahms** [comment]
@pieterdc1 appears to be fixed in 0.87.0? https://semgrep.dev/s/ZPZ4

**Comment by mjambon** [comment]
Update: still needs fixing.
